impeller/entity/geometry/line_geometry.cc

```diff
@@ -0,0 +1,147 @@
+                                               const Entity& entity,
+                                               RenderPass& pass) {
+  auto& host_buffer = pass.GetTransientsBuffer();
+
+  Point corners[4];
+  if (!ComputeCorners(corners, entity.GetTransformation(),
+                      cap_ == Cap::kSquare)) {
+    return {};
+  }
+
+  return GeometryResult{
+      .type = PrimitiveType::kTriangleStrip,
+      .vertex_buffer =
+          {
+              .vertex_buffer = host_buffer.Emplace(corners, 8 * sizeof(float),
+                                                   alignof(float)),
+              .vertex_count = 4,
+              .index_type = IndexType::kNone,
+          },
+      .transform = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransformation(),
+      .prevent_overdraw = false,
+  };
+}
+
+// |Geometry|
+GeometryResult LineGeometry::GetPositionUVBuffer(Rect texture_coverage,
+                                                 Matrix effect_transform,
+                                                 const ContentContext& renderer,
+                                                 const Entity& entity,
+                                                 RenderPass& pass) {
+  auto& host_buffer = pass.GetTransientsBuffer();
+
+  auto uv_transform =
+      texture_coverage.GetNormalizingTransform() * effect_transform;
+  Point corners[4];
+  if (!ComputeCorners(corners, entity.GetTransformation(),
+                      cap_ == Cap::kSquare)) {
+    return {};
+  }
+
+  std::vector<Point> data(8);
+  for (auto i = 0u, j = 0u; i < 8; i += 2, j++) {
+    data[i] = corners[j];
+    data[i + 1] = uv_transform * corners[j];
+  }
+
+  return GeometryResult{
+      .type = PrimitiveType::kTriangleStrip,
+      .vertex_buffer =
+          {
+              .vertex_buffer = host_buffer.Emplace(
+                  data.data(), 16 * sizeof(float), alignof(float)),
+              .vertex_count = 4,
+              .index_type = IndexType::kNone,
+          },
+      .transform = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransformation(),
+      .prevent_overdraw = false,
+  };
+}
+
+GeometryVertexType LineGeometry::GetVertexType() const {
+  return GeometryVertexType::kPosition;
+}
+
+std::optional<Rect> LineGeometry::GetCoverage(const Matrix& transform) const {
+  Point corners[4];
+  if (!ComputeCorners(corners, transform, cap_ != Cap::kButt)) {
+    return {};
+  }
+
+  for (int i = 0; i < 4; i++) {
+    corners[i] = transform * corners[i];
+  }
+  return Rect::MakePointBounds(std::begin(corners), std::end(corners));
+}
+
+bool LineGeometry::CoversArea(const Matrix& transform, const Rect& rect) const {
+  if (!transform.IsTranslationScaleOnly() || !IsAxisAlignedRect()) {
+    return false;
+  }
+  auto coverage = GetCoverage(transform);
+  return coverage.has_value() ? coverage->Contains(rect) : false;
+}
+
+bool LineGeometry::IsAxisAlignedRect() const {
```

Comment:
I think if you're drawing a line, its only an axis aligned rect if it is a square cap?

Reply:
square or butt, I'll include this in the issue for handling round caps...